src/develop/imageop_gui.c

```diff
@@ -0,0 +1,219 @@
+
+    float top = fmaxf(fabsf(min),fabsf(max));
+    if (top>=100)
+    {
+      step = 1.f;
+    }
+    else
+    {
+      step = top / 100;
+      float log10step = log10f(step);
+      float fdigits = floorf(log10step+.1);
+      step = powf(10.f,fdigits);
+      if (log10step - fdigits > .5)
+        step *= 5;
+      if (fdigits < -2.f) 
+        digits = -fdigits;
+    }
+
+    slider = dt_bauhaus_slider_new_with_range_and_feedback(self, min, max, step, defval, digits, 1);
+
+    if (*f->header.description)
+    {
+      dt_bauhaus_widget_set_label(slider, NULL, gettext(f->header.description));
+    }
+    else
+    {
+      str = dt_util_str_replace(f->header.field_name, "_", " ");
+    
+      dt_bauhaus_widget_set_label(slider, NULL, gettext(str));
+
+      g_free(str);
+    }
+
+    const char *post = ""; // set " %%", " EV" etc
+
+    if (min < 0 || (post && *post))
+    {
+      str = g_strdup_printf("%%%s.0%df%s", (min < 0 ? "+" : ""), digits, post);
+
+      dt_bauhaus_slider_set_format(slider, str);
+    
+      g_free(str);
+    }
+
+    dt_module_param_t *module_param = (dt_module_param_t *)g_malloc(sizeof(dt_module_param_t));
+    module_param->module = self;
+    module_param->param_offset = self->so->get_p(p, param) - p;
+    g_signal_connect_data(G_OBJECT(slider), "value-changed", G_CALLBACK(generic_slider_callback), module_param, (GClosureNotify)g_free, 0);
+  }
+  else
+  {
+    str = g_strdup_printf("'%s' is not a float/slider parameter", param);
+
+    slider = GTK_WIDGET(GTK_LABEL(gtk_label_new(str)));
+
+    g_free(str);
+  }
+
+  gtk_box_pack_start(GTK_BOX(self->widget), slider, FALSE, FALSE, 0);
+
+  return slider;
+}
+
+GtkWidget *dt_bauhaus_combobox_new_from_params_box(dt_iop_module_t *self, const char *param)
+{
+  dt_iop_params_t *p = (dt_iop_params_t *)self->params;
+  dt_introspection_field_t *f = self->so->get_f(param);
+
+  GtkWidget *combobox = NULL;
+  gchar *str;
+
+  if (f && (f->header.type == DT_INTROSPECTION_TYPE_ENUM || f->header.type == DT_INTROSPECTION_TYPE_INT))
+  {
+    combobox = dt_bauhaus_combobox_new(self);
+
+    if (*f->header.description)
+    {
+      dt_bauhaus_widget_set_label(combobox, NULL, gettext(f->header.description));
+    }
+    else
+    {
+      str = dt_util_str_replace(f->header.field_name, "_", " ");
+    
+      dt_bauhaus_widget_set_label(combobox, NULL, gettext(str));
+
+      g_free(str);
+    }
+
+    dt_module_param_t *module_param = (dt_module_param_t *)g_malloc(sizeof(dt_module_param_t));
+    module_param->module = self;
+    module_param->param_offset = self->so->get_p(p, param) - p;
+    g_signal_connect_data(G_OBJECT(combobox), "value-changed", G_CALLBACK(generic_combobox_callback), module_param, (GClosureNotify)g_free, 0);
+
+    if (f->header.type == DT_INTROSPECTION_TYPE_ENUM)
+    {
+     for(dt_introspection_type_enum_tuple_t *iter = f->Enum.values; iter->name; iter++)
+      {
+//        dt_bauhaus_combobox_add(combobox, gettext(iter->description));
```

Comment:
Is that expected that no entry are added into the combobox ? If ok and code for later please add a comment otherwise uncomment this line.

Reply:
added brief comment in the code saying @houz is working on this

Reply:
Ok, so I better let @houz review and give the go for this.